modules/react-components/src/components/header/header.tsx

```diff
@@ -215,14 +215,20 @@ export const Header: FunctionComponent<HeaderPropsInterface> = (
     };
 
     const resolveAuthenticatedUserEmail = (): string => {
-        if (!profileInfo || !profileInfo.emails || !profileInfo.emails.length || profileInfo.emails.length < 1) {
+        if (!profileInfo || !profileInfo.userName) {
             return null;
         }
 
-        if (typeof profileInfo.emails[ 0 ] === "string") {
-            return profileInfo.emails[ 0 ];
-        } else if (typeof profileInfo.emails[ 0 ] === "object") {
-            return profileInfo.emails[ 0 ].value;
+        if (typeof profileInfo.userName === "string") {
+            const userName = profileInfo?.userName?.split("/").length > 1
+            ? profileInfo?.userName?.split("/")[1]
+            : profileInfo?.userName;
```

Comment:
if `profileInfo.userName` is undefined, the function will be returned from line number `219`.So optional chaining here is redundant.
```suggestion
            : profileInfo.userName;
```